jaeger-core/src/main/java/io/jaegertracing/internal/senders/SenderResolver.java

```diff
@@ -55,7 +55,7 @@ public static Sender resolve(Configuration.SenderConfiguration senderConfigurati
     Iterator<SenderFactory> senderFactoryIterator = senderFactoryServiceLoader.iterator();
 
     boolean hasMultipleFactories = false;
-    if (senderFactoryIterator.hasNext()) {
+    while (senderFactoryIterator.hasNext()) {
```

Comment:
Found this bug while testing the `NoopSenderFactory` and left the test as `testNoopSenderFactory`.

Reply:
No, the original `if` was fine I think? it is just checking whether the is more than one - it doesn't need to iterate through them all.

Reply:
The check comes a few lines later. This is really meant to be a `while`, the test I added will fail when this line is `if`, as proof :)

Reply:
Ah yes - I was still thinking the property specified the class, but its based on matching the type, so does need to iterate.